eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/serialization/commons/NestedBundle.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2017 SchedJoules
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.schedjoules.eventdiscovery.framework.serialization.commons;
+
+import android.content.Intent;
+import android.os.Bundle;
+
+import com.schedjoules.eventdiscovery.framework.serialization.Keys;
+import com.schedjoules.eventdiscovery.framework.utils.factory.Factory;
+import com.schedjoules.eventdiscovery.framework.utils.optionals.AbstractCachingOptional;
+
+import org.dmfs.optional.NullSafe;
+import org.dmfs.optional.Optional;
+
+
+/**
+ * @author Gabor Keszthelyi
+ */
+public final class NestedBundle extends AbstractCachingOptional<Bundle>
+{
+    public NestedBundle(final Intent intent)
+    {
+        super(new Factory<Optional<Bundle>>()
+        {
+            @Override
+            public Optional<Bundle> create()
+            {
+                return new NullSafe<>(intent.getBundleExtra(Keys.NESTED_BUNDLE));
```

Comment:
> new NestedBundle(getIntent()).value(null);

This was only replicating the previous usage (`getIntent().getExtras()`) where it could be null as well, but I agree it's better to always provide a Bundle arg for a Fragment which can have arguments.

>```
><T> BuilderAdapter withExtra(Key<T> key, Optional<T> value);
>```

We have `Key<T>` and `Box<T>` as parameters, we can't use general `<T>` or `Optional<T>` cause we don't know how to box it.

So that's why this won't work either:
> ```
> return new IntentBuilder(intent).withExtra(Keys.THEME, new Extra<>(Keys.THEME, getIntent())).build();
> ```

But I've add a `withCopiedExtra(Key<T> key, Intent intent)` to the builder, that works.

Btw, `NestedBundle` should really not be used normally, that's one aim of this design is to hide that boilerplate. It's internal to how we put extras into Intent. The usage in `EventListActivity` was only needed to take out the intent extras as forward as fragment bundle.

So I've added the initial version for the builders, please check them out.

I will write to you on Slack about the Obligatory and Optional usage topic.